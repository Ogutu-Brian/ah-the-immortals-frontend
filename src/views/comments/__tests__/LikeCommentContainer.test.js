import React from 'react';
import { Router } from 'react-router-dom';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { createBrowserHistory } from 'history';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import toJson from 'enzyme-to-json';
import LikeComment from '../LikeCommentContainer';

describe('Tests LikeCommentContainer', () => {
  const store = configureStore([thunk])({
    signin: {
      currentUser: 'issa', 
    },
  });
  const props = {
    comment: {
      'id': 87,
      'createdAt': '2018-06-25',
      'updatedAt': '2018-06-25',
      'body': 'string-tests',
      'author': {
        'username': 'issa',
        'bio': '',
        'image': 'wqout',
        'following': false
      },
      'parent': null,
      'likesInfo': {
        'like': false,
        'dislike': false,
        'like_count': 0,
        'dislikes_count': 0
      }},
    isAuthenticated: 'issa',
    likeCommentAction: jest.fn(),
    slug: 'string'
  };
  const history = createBrowserHistory();
  const wrapper = mount (
    <Router history={history}>
      <Provider store={store}>
        <LikeComment {...props} />
      </Provider>
    </Router>
  );
  it('tests mounting of container', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('tests clicking of the like button', () => {
    const like = wrapper.find('i[id="like"]');
    like.simulate('click');
    expect(like.exists()).toEqual(true);
    expect(like.length).toEqual(1);
  });
});
