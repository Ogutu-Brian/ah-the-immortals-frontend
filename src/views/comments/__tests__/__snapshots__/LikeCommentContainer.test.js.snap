// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests LikeCommentContainer tests mounting of container 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Connect(LikeComment)
      comment={
        Object {
          "author": Object {
            "bio": "",
            "following": false,
            "image": "wqout",
            "username": "issa",
          },
          "body": "string-tests",
          "createdAt": "2018-06-25",
          "id": 87,
          "likesInfo": Object {
            "dislike": false,
            "dislikes_count": 0,
            "like": false,
            "like_count": 0,
          },
          "parent": null,
          "updatedAt": "2018-06-25",
        }
      }
      isAuthenticated="issa"
      likeCommentAction={[MockFunction]}
      slug="string"
    >
      <LikeComment
        comment={
          Object {
            "author": Object {
              "bio": "",
              "following": false,
              "image": "wqout",
              "username": "issa",
            },
            "body": "string-tests",
            "createdAt": "2018-06-25",
            "id": 87,
            "likesInfo": Object {
              "dislike": false,
              "dislikes_count": 0,
              "like": false,
              "like_count": 0,
            },
            "parent": null,
            "updatedAt": "2018-06-25",
          }
        }
        isAuthenticated="issa"
        likeCommentAction={[Function]}
        promptAction={[Function]}
        slug="string"
      >
        <LikeDislikeComment
          comment={
            Object {
              "author": Object {
                "bio": "",
                "following": false,
                "image": "wqout",
                "username": "issa",
              },
              "body": "string-tests",
              "createdAt": "2018-06-25",
              "id": 87,
              "likesInfo": Object {
                "dislike": false,
                "dislikes_count": 0,
                "like": false,
                "like_count": 0,
              },
              "parent": null,
              "updatedAt": "2018-06-25",
            }
          }
          handleClick={[Function]}
          isAuthenticated="issa"
        >
          <div
            className="commentlike "
          >
            <div>
              <i
                aria-hidden="true"
                className="fas fa-thumbs-up fa-1x like-icon left-icon"
                comment={
                  Object {
                    "author": Object {
                      "bio": "",
                      "following": false,
                      "image": "wqout",
                      "username": "issa",
                    },
                    "body": "string-tests",
                    "createdAt": "2018-06-25",
                    "id": 87,
                    "likesInfo": Object {
                      "dislike": false,
                      "dislikes_count": 0,
                      "like": false,
                      "like_count": 0,
                    },
                    "parent": null,
                    "updatedAt": "2018-06-25",
                  }
                }
                id="like"
                onClick={[Function]}
                style={
                  Object {
                    "color": "#909090",
                  }
                }
              />
            </div>
            <div
              className="likes-count "
            >
              0
            </div>
            <div>
              <i
                aria-hidden="true"
                className="fas fa-thumbs-up fa-rotate-180 fa-1x like-icon right-icon"
                id="dislike"
                onClick={[Function]}
                style={
                  Object {
                    "color": "#909090",
                  }
                }
              />
            </div>
            <div
              className="likes-count "
            >
              0
            </div>
          </div>
        </LikeDislikeComment>
      </LikeComment>
    </Connect(LikeComment)>
  </Provider>
</Router>
`;
